targets/TARGET_STM/TARGET_STM32L4/TARGET_MTS_DRAGONFLY_L471QG/PinNames.h

```diff
@@ -0,0 +1,399 @@
+	PD_6 = 0x36,
+	PD_7 = 0x37,
+	PD_8 = 0x38,
+	PD_9 = 0x39,
+	PD_10 = 0x3A,
+	PD_11 = 0x3B,
+	PD_12 = 0x3C,
+	PD_13 = 0x3D,
+	PD_14 = 0x3E,
+	PD_15 = 0x3F,
+
+	PE_0 = 0x40,
+	PE_1 = 0x41,
+	PE_2 = 0x42,
+	PE_3 = 0x43,
+	PE_4 = 0x44,
+	PE_5 = 0x45,
+	PE_6 = 0x46,
+	PE_7 = 0x47,
+	PE_8 = 0x48,
+	PE_9 = 0x49,
+	PE_10 = 0x4A,
+	PE_11 = 0x4B,
+	PE_12 = 0x4C,
+	PE_13 = 0x4D,
+	PE_14 = 0x4E,
+	PE_15 = 0x4F,
+
+	PF_0 = 0x50,
+	PF_1 = 0x51,
+	PF_2 = 0x52,
+	PF_3 = 0x53,
+	PF_4 = 0x54,
+	PF_5 = 0x55,
+	PF_6 = 0x56,
+	PF_7 = 0x57,
+	PF_8 = 0x58,
+	PF_9 = 0x59,
+	PF_10 = 0x5A,
+	PF_11 = 0x5B,
+	PF_12 = 0x5C,
+	PF_13 = 0x5D,
+	PF_14 = 0x5E,
+	PF_15 = 0x5F,
+
+	PG_0 = 0x60,
+	PG_1 = 0x61,
+	PG_2 = 0x62,
+	PG_3 = 0x63,
+	PG_4 = 0x64,
+	PG_5 = 0x65,
+	PG_6 = 0x66,
+	PG_7 = 0x67,
+	PG_8 = 0x68,
+	PG_9 = 0x69,
+	PG_10 = 0x6A,
+	PG_11 = 0x6B,
+	PG_12 = 0x6C,
+	PG_13 = 0x6D,
+	PG_14 = 0x6E,
+	PG_15 = 0x6F,
+
+	PH_0 = 0x70,
+	PH_1 = 0x71,
+
+	// ADC internal channels
+	ADC_TEMP = 0xF0,
+	ADC_VREF = 0xF1,
+	ADC_VBAT = 0xF2,
+
+	// Arduino connector namings
+	A0  = PC_2,
+	A1  = PC_0,     //rev b PC_0; rev c PC_13
+	A2  = PC_4,
+	A3  = PB_1,     //reb b PB_1; rev c PE_6
+	A4  = PA_6,
+	A5  = PG_8,
+	D0  = PA_3,
+	D1  = PA_2,
+	D2  = PB_15,
+	D3  = PA_0,
+	D4  = PA_7,
+	D5  = PA_9,
+	D6  = PA_1,
+	D7  = PG_7,
+	D8  = PB_0,
+	D9  = PB_10,
+	D10 = PC_8,
+	D11 = PB_5,
+	D12 = PG_3,
+	D13 = PG_2,
+	D14 = PB_9,
+	D15 = PB_8,
+
+	// 40 pin
+	IO_00 = D1,
+	IO_01 = D4,
+	IO_02 = D8,
+	IO_03 = D6,
+	IO_04 = D11,
+	IO_05 = D13,
+	IO_06 = D15,
+	IO_07 = D2,
+	IO_08 = A0,
+	IO_09 = A3,
+	IO_10 = A1,
+	IO_11 = A4,
+	IO_12 = A2,
+	IO_13 = D9,
+	IO_14 = A5,
+	IO_15 = D14,
+	IO_16 = D12,
+	IO_17 = D10,
+	IO_18 = D3,
+	IO_19 = D5,
+	IO_20 = D7,
+	IO_21 = D0,
+
+	// 40 pin USB debug port
+	USBTX = PB_6,
+	USBRX = PB_7,
+
+	//UARTS
+	UART3_TX = PD_2,
+	UART3_RX = PD_9,
+	UART3_CTS = PD_11,
+	UART3_RTS = PD_12,
+
+	UART2_TX = PA_2,
+	UART2_RX = PA_3,
+	UART2_RTS = PA_1,
+	UART2_CTS = PA_0,
+	UART2_DSR = PA_9,
+	UART2_DTR = PG_7,
+	UART2_DCD = PA_7,
+
+
+	UART1_TX = PB_6,
+	UART1_RX = PB_7,
+	UART1_RTS = PA_12,
+	UART1_CTS = PA_11,
+
+	// 40 pin JTAG/SWD
+	J_TCK = PA_14,	// a.k.a. SWCLK
+	J_TDI = PA_15,
+	J_RST = PB_4,
+	J_TDO = PB_3,	// a.k.a. SWO
+	J_TMS = PA_13,	// a.k.a. SWDIO
+
+	// Generic signals namings
+	LED1 = D3,
+	LED2 = D3,
+	LED3 = D3,
+	LED4 = D3,
+
+	// SERIAL PORT
+    SERIAL_TX  = D1,
+    SERIAL_RX  = D0,
+    SERIAL_RTS = D6,
+    SERIAL_CTS = D3,
+    SERIAL_DCD = D4,
+    SERIAL_DSR = D5,
+    SERIAL_DTR = D7,
+    SERIAL_RI  = D8,
+
+	// SPI1 and SPI2 are available on Arduino pins
+    SPI1_MOSI = D11,
+    SPI1_MISO = D12,
+    SPI1_SCK  = D13,
+    SPI2_MOSI = D2,
+    SPI2_MISO = A0,
+    SPI2_SCK  = D9,
+
+    // SPI3 connects to flash part
+	SPI3_MOSI = PC_12,
+	SPI3_MISO = PC_11,
+	SPI3_SCK  = PC_10,
+	SPI3_SSEL = PG_12,
+
+    // I2C1 and I2C3 are available on Arduino pins
+    I2C1_SCL = D15,
+    I2C1_SDA = D14,
+    I2C3_SCL = D7,
+    I2C3_SDA = A5,
+
```

Comment:
@cedrickkukela-cd : Can you please define defaults for each peripheral? I see there are multiple UART's, SPI's etc,

Reply:
Im not quite sure i understand the request here. Could you elaborate.

Reply:
You can create aliases for I2C_SDA = I2C1_SDA (or I2C3_SDA based on whichever you wish to make as default) (& similarly for all other peripherals) so that application code can invoke the peripheral & create the object with just I2C_SDA and not have to choose I2C1_SDA / I2C3_SDA / any other..